app/workers/bulk_migrate_list_worker.rb

```diff
@@ -0,0 +1,48 @@
+class BulkMigrateListWorker < ApplicationWorker
+  def perform(source_list_id, successor_list_id)
+    run_with_advisory_lock(SubscriberList, source_list_id) do
+      source_list = SubscriberList.find(source_list_id)
+      successor_list = SubscriberList.find(successor_list_id)
+      source_subscriptions = Subscription.active.find_by(subscriber_list_id: source_list.id)
+
+      if source_subscriptions.nil?
+        logger.warn("no active subscriptions for #{source_list.title}")
+        return
+      end
+
+      migrate_subscribers(source_list, successor_list)
+    end
+  end
+
+private
+
+  def migrate_subscribers(source_list, successor_list)
```

Comment:
Yes, I intended to update the commit message to explain this. They were only small reasons:

- SubscriberListMover has multiple `puts`'s instead of writing to rails logger, and tbh I wasn't sure what that would do/where it would output. 
- Much of the code isn't useful here, the sending of bulk email and the guard clauses (which we are handling in the controller). So I thought I'd refactor the SubscriberListMover (call an email builder class for example). But then I changed my mind because this endpoint is temporary, i.e. maybe it was easier to add all of the code and new specs in one PR to make it easier to follow, and then revert as a whole.

None of these are very strong reasons tho, so I can refactor if you think I should.

Reply:
Are you happy for me to merge Leena, or do you think refactor is needed?

Reply:
I think it's fine for now. If it looks like this code is going to become permanent, then I think it should be refactored then.